pkg/sessions/redis/redis_store.go

```diff
@@ -54,6 +54,30 @@ func (store *SessionStore) Load(ctx context.Context, key string) ([]byte, error)
 	return value, nil
 }
 
+// Load reads sessions.SessionState information from a persistence
+// cookie within the HTTP request object and locks it on redis
+func (store *SessionStore) LoadWithLock(ctx context.Context, key string) ([]byte, error) {
+	value, err := store.Client.Get(ctx, key)
+	if err != nil {
+		return nil, fmt.Errorf("error loading redis session: %v", err)
+	}
+
+	err = store.Client.Lock(ctx, key, 300*time.Millisecond)
```

Comment:
This potentially feels too short.

We don't have data, but we'd ideally want P90 or P99 response times to IdP refresh endpoints.  I have no idea what that would be though... 🤔 

1s feels right (it normally releases earlier manually when the operation was faster, right?)